spec/pry_repl_spec.rb

```diff
@@ -136,4 +136,40 @@ def output.tty?; true; end
       end
     end
   end
+
+  describe '#calculate_overhang' do
+    it 'is called' do
+      expect_any_instance_of(Pry::REPL)
+        .to receive(:calculate_overhang)
+        .with(' x = 1', 'x = 1')
+        .and_return(1)
+      # needed to exit repl thread
+      expect_any_instance_of(Pry::REPL)
+        .to receive(:calculate_overhang)
+        .with('exit-all', 'exit-all')
+        .and_return(0)
+      ReplTester.start(correct_indent: true, auto_indent: true) do
+        output=@pry.config.output
+        def output.tty?; true; end
+        input ' x = 1'
+      end
+    end
+
+    it 'uses indented_val if not readline' do
+      ReplTester.start(correct_indent: true, auto_indent: true) do
+        def @repl.readline_available?; false; end
+        overhang = @repl.send(:calculate_overhang, '  foo', 'foo')
+        raise "Test failed" unless overhang == 2
```

Comment:
If you tell me how can I use `expect()` inside this block, I would be happy :)

Reply:
Well the tests as they are do test the functionality, they are just not very nice implemention-wise (the `raise "test failse"`). But I will remove them if you wish so.

Reply:
Yeah, let's just remove them. The current tests are very brittle because of mocking and querying private API.